code/client/src/pages/Show/Swap.jsx

```diff
@@ -0,0 +1,268 @@
+  textAlign: 'left',
+  height: '50px'
+}
+
+const selectOptionStyle = {
+  padding: 0
+}
+
+const maxButtonStyle = {
+  marginLeft: '20px',
+  bottom: '1px'
+}
+
+const amountInputStyle = {
+  margin: 0
+}
+
+/**
+ * Renders swap coins from ONE wallet or tracked token to another token tab.
+ */
+const Swap = ({ address }) => {
+  const wallets = useSelector(state => state.wallet.wallets)
+  const network = useSelector(state => state.wallet.network)
+  const wallet = wallets[address] || {}
+  const tokenBalances = wallet.tokenBalances || {}
+  const trackedTokens = (wallet.trackedTokens || []).filter(e => e.tokenType === ONEConstants.TokenType.ERC20)
+  const harmonyToken = {
+    name: HarmonyONE.name,
+    icon: HarmonyONE.icon,
+    symbol: HarmonyONE.symbol
+  }
+  const harmonySelectOption = {
+    ...harmonyToken,
+    value: HarmonyONE.symbol,
+    label: <TokenLabel token={harmonyToken} />
+  }
+  const defaultTrackedTokens = withKeys(DefaultTrackedERC20(network))
+  const [currentTrackedTokens, setCurrentTrackedTokens] = useState([harmonyToken, ...defaultTrackedTokens, ...(trackedTokens || [])])
+  const balances = useSelector(state => state.wallet.balances)
+  const balance = balances[address] || 0
+  const [supportedTokens, setSupportedTokens] = useState([])
+  const [selectedTokenSwapFrom, setSelectedTokenSwapFrom] = useState(harmonySelectOption)
+  const [selectedTokenSwapTo, setSelectedTokenSwapTo] = useState({ value: '', label: '' })
+  const [swapAmount, setSwapAmount] = useState()
+  const [selectedTokenBalance, setSelectedTokenBalance] = useState('0')
+
+  // Loads supported tokens that are available for swap.
+  useEffect(() => {
+    const loadCryptos = async () => {
+      // TODO: this is not tokens supported by Harmony network, fetch supported tokens from Harmony network.
+      const loadedCryptos = await mockCryptos()
+      setSupportedTokens(loadedCryptos)
+    }
+
+    loadCryptos()
+  }, [setSupportedTokens])
+
+  useEffect(() => {
+    const loadTrackedTokensMetadata = async () => {
+      const trackedTokensWithMetadata = await Promise.all(currentTrackedTokens.map(async (trackedToken) => {
+        try {
+          if (trackedToken.symbol === HarmonyONE.symbol) {
+            return trackedToken
+          }
+
+          const { name, symbol, decimals } = await api.blockchain.getTokenMetadata(trackedToken)
+
+          return {
+            ...trackedToken,
+            name,
+            symbol,
+            decimals
+          }
+        } catch (ex) {
+          console.error(ex)
+        }
+      }))
+      setCurrentTrackedTokens(trackedTokensWithMetadata)
+    }
+
+    loadTrackedTokensMetadata()
+  }, [currentTrackedTokens, setCurrentTrackedTokens])
+
+  useEffect(() => {
+    const tokenBalance = getSelectedTokenComputedBalance(selectedTokenSwapFrom, tokenBalances, balance)
+
+    if (!tokenBalance) {
+      return '0'
+    }
+
+    setSelectedTokenBalance(tokenBalance.formatted)
```

Comment:
Should set raw amount and display formatted. Transactions would rely on raw amounts

Reply:
`selectedTokenBalance` is used in input `value` for display, so it is a formatted value. When submit, the value is converted to raw amount.